frontend/src/concepts/pipelines/topology/core/PipelineVisualizationSurface.tsx

```diff
@@ -22,23 +24,39 @@ const PipelineVisualizationSurface: React.FC<PipelineVisualizationSurfaceProps>
   selectedIds,
 }) => {
   const controller = useVisualizationController();
-
+  const [error, setError] = React.useState<Error | null>();
   React.useEffect(() => {
     // PF Bug
     // TODO: Pipeline Topology weirdly doesn't set a width and height on spacer nodes -- but they do when using finally spacer nodes
     const spacerNodes = getSpacerNodes(nodes).map((s) => ({ ...s, width: 1, height: 1 }));
     const renderNodes = [...spacerNodes, ...nodes];
     // TODO: We can have a weird edge issue if the node is off by a few pixels vertically from the center
     const edges = getEdgesFromNodes(renderNodes);
-
-    controller.fromModel(
-      {
-        nodes: renderNodes,
-        edges,
-      },
-      true,
-    );
+    try {
+      controller.fromModel(
+        {
+          nodes: renderNodes,
+          edges,
+        },
+        true,
+      );
+    } catch (error) {
+      if (error instanceof Error) {
+        setError(error);
+      }
+    }
```

Comment:
Can you maybe log something here if it's not somehow an instance of Error (which I imagine everything will be).

```suggestion
    } catch (error) {
      if (error instanceof Error) {
        setError(error);
      } else {
        // eslint-disable-next-line no-console
        console.error('Unknown error occurred rendering Pipeline Graph', error);
      }
    }
```